### ISSUE TITLE
Unable to download files

### ISSUE THREAD

**Issue by karanrajpal14** [fabe/gatsby-plugin-drive#12]
Hello,

I've currently set up my config like so:
```
{
      resolve: '@fs/gatsby-plugin-drive',
      options: {
        folderId: '1EHtphzknN_vt9QtVjl1AziHu2_p0VEYL',
        keyFile: path.resolve(__dirname, 'client_secret.json'),
        destination: path.join(__dirname, 'src/content'),
        exportGDocs: true,
        exportMimeType: 'application/pdf',
      }
    },
```

I've got two documents in that folder that I'd like to use. And I have shared those docs with the specified the email but the download seems to be failing. Here's the error:

```
🚗  Started downloading content
⠸ onPreBootstrap

 ERROR 

UNHANDLED REJECTION filesToDownload is not iterable



  TypeError: filesToDownload is not iterable
  
  - gatsby-node.js:42 
    [karanr.dev]/[@fs]/gatsby-plugin-drive/gatsby-node.js:42:22
  
  - new Promise
  
  - gatsby-node.js:38 recursiveFolders
    [karanr.dev]/[@fs]/gatsby-plugin-drive/gatsby-node.js:38:10
  
  - gatsby-node.js:30 
    [karanr.dev]/[@fs]/gatsby-plugin-drive/gatsby-node.js:30:5
  
  - task_queues.js:93 processTicksAndRejections
    internal/process/task_queues.js:93:5
```

Any idea what's going wrong?

**Comment by vbodell** [comment]
+1. 
From what I can tell the array should be the result of: `await googleapi.getFolder(folderId, token);` but it seems as if the call doesn't return any files/an array. My guess is that `googleapi.getFolder()` returns an error due to some googleApi setup that we both have, which isn't handled in `onPreBootstrap`. If that's the case then it might be nice to get the error message in the console rather than the `TypeError`. But I could be off on this.

**Comment by stcalica** [comment]
Looking in his GoogleAPIs code (src/googleapis.js) he has: 
```
const getFolder = (folderId, token) => {
  return new Promise((resolve, reject) => {
    request(
      {
        uri: `https://www.googleapis.com/drive/v3/files`,
        auth: {
          bearer: token,
        },
        qs: {
          q: `'${folderId}' in parents`,
        },
      },
      (err, res, body) => {
        if (err) {
          reject(err);
        } else {
          resolve(JSON.parse(body).files);
        }
      }
    );
  });
};
```
when I removed this: 

```        qs: {
          q: `'${folderId}' in parents`,
        },
      },
```

I was able to get the files. 

I don't know what that query string is supposed to do but it's not right. 

I'm using Postman and it doesn't even work there. 
I suggest we remove this and we can get the files we want out.

**Comment by stcalica** [comment]
Fixed that issue but now the recursion is all wrong. 
He is not responding it seems. 
I might take this project as a fork in my own repo and make the needed adjustments as my team needs this ASAP

<img width="950" alt="Screen Shot 2020-03-09 at 1 08 17 PM" src="https://user-images.githubusercontent.com/5624835/76253000-0d8f2700-6207-11ea-8947-d47d0fdbcf94.png">

**Comment by karanrajpal14** [comment]
@stcalica do let me know if you have a fork that's working. I'd like to give it a whirl as well.

**Comment by stcalica** [comment]
@karanrajpal14  I'll have to get around to it. 
I'll need to figure out why recursion was needed and how to filter out other files. 
Also, another limitation is the `export` endpoint of Google Drive seems to have a limitation of 10MB. 
And I need to pull some videoes. If I can't get that working then I'll have to use something else for my project and put this on the backburner. Will update what I'm doing this week!

**Comment by fabe** [comment]
@stcalica If I remember correctly, recursion is needed because folders are treated as files. So for each folder, we need to fetch all of the files and folders contained in it, and so on. Hence the `in parents` query.

I'd be happy to review a pull request and publish a new version, unfortunately I don't have time at the moment to work on this myself.

**Comment by stcalica** [comment]
@fabe  yeah I think they changed their API from when this was written. I can't find a query similar to that anymore in updated docs. 
When I ran mine it showed everything flatten. As it showed all my files even ones nested in the folders and then the folders as files. We might just need to remove the folders now since it seems to extract all. But need to do more investigation as my testing directory was large.

**Comment by karanrajpal14** [comment]
@stcalica how have you decided to move ahead with this? I'm thinking on just patching the small change in my local modules and moving ahead with it since I don't really have a very complex folder structure.

**Comment by stcalica** [comment]
@karanrajpal14 yeah I'd do that for now. Might tackle it as one of my "quarantine" goals. I ended up using something else to store my photos and abandoned videos for now since Google Drive won't even store them anyway :/

**Comment by karanrajpal14** [comment]
@stcalica looks like I'm going to be poking around this during the quarantine as well. Mind sharing the `src/googleapis.js` as a whole after you made the changes? I tried making the change you mentioned and it did not work for me. Could you share your changes so that I can try them out?

Also, what are you using to fetch the photos from Drive?

**Comment by stcalica** [comment]
@karanrajpal14  
My changes were to remove the query string only. 
From above: 
https://github.com/fabe/gatsby-plugin-drive/issues/12#issuecomment-596751582

I'll be playing around with this next week just to help a bit. 
But google drive was not the tool for me.

**Comment by loongyh** [comment]
I have encountered this issue after changing the sharing settings on one of the GDrive `folderId` subfolders. My local `netlify dev` setup worked properly however this error was thrown when deploying on Netlify.

At first I thought there was some issue with my latest commit however I tried redeploying an older successful build and it failed too.

Finally fixed it by removing and recreating the GDrive folder with a different `folderId`, and updating the plugin's `gatsby-config.js` settings accordingly.
